editor/src/components/canvas/controls/select-mode/flex-gap-control.tsx

```diff
@@ -130,21 +132,40 @@ export const FlexGapControl = controlForStrategyMemoized<FlexGapControlProps>((p
     elementPathTrees,
     selectedElement,
   )
-  const flexGap = maybeFlexGapData(metadata, selectedElement)
-  if (flexGap == null) {
-    return null
-  }
 
-  const flexGapValue = updatedGapValue ?? flexGap.value
+  const flexGapFromEditor = useEditorState(
```

Comment:
the refactor here was necessary because the early return didn't respect the rules of hooks